### DISCUSSION TITLE
UI improvements

### DISCUSSION THREAD

**Discussion by stereobooster** [jpanther/congo#651] [Ideas]
Note: this is all **highly subjective**. I'd rather want to discuss those idea. If you think they make sense I can make separate issues/pull requests.

## Collapsible header

Links in the header gets corrupted on small screens:

<img width="738" alt="Screenshot 2023-09-12 at 22 55 11" src="https://github.com/jpanther/congo/assets/179534/561a6f55-255a-4e46-b9ca-21865585ef05">

One of possible solution is to use something similar to [Bulma](https://bulma.io/)

<img width="843" alt="Screenshot 2023-09-12 at 23 01 45" src="https://github.com/jpanther/congo/assets/179534/bd73382d-3de5-470b-881f-1f571a2d898d">
<img width="628" alt="Screenshot 2023-09-12 at 23 01 53" src="https://github.com/jpanther/congo/assets/179534/9ec02427-2a9c-4cbd-8cad-8c378d848897">

## Kbd can look more like keyboard button

For example in github <kbd>Enter</kbd>. And in [tailwind](https://github.com/tailwindlabs/tailwindcss-typography/pull/317)

## Remove quotes from `blockquote`

Because when I add citation like this

> quote
>
> -- author

It doesn't make sense

<img width="433" alt="Screenshot 2023-09-12 at 23 06 47" src="https://github.com/jpanther/congo/assets/179534/8320a5bd-51f6-4e5a-9dab-877b0ca1f48a">

## Support line numbers in code-blocks

In order to do this need to adjust styles/scripts a bit:

- line numbers should not be selectable (`user-select: none;`)
- line numbers should not be in code if user uses copy button
- ideally line numbers should have fixed position (if there is horizontal scroll)

## Scroll observer for ToC

So when user scrolls they know at which section they are

**Comment by jpanther** [comment]
I generally like these suggestions, with the following comments...

- Collapsible header would be a useful addition, perhaps with some way of prioritising what gets collapsed. The current header partials could use a bit of an overhaul in terms of reducing code repetition and making them more flexible.
- The theme already has specific styles for kbd tags so I don't think this is an issue, or do you not like the current styling?
- Removing quotes from block quotes makes sense also, although I feel like that's more an upstream issue with Tailwind typography
- Scroll observer for ToC is something that's been on my todo list for a long time so I'd love a simple implementation for this. Ideally I don't want to include a bunch of third party code. The ToC system in Hugo is extremely limited and so it's not easy to work with and this has been my hurdle to making a better styled template.

**Reply by stereobooster** [reply]
For TOC - maybe not easiest to implement, but pretty cool https://kld.dev/toc-animation/. [I implemented it for Astro](https://astro-digital-garden.stereobooster.com/recipes/table-of-contents/#snake-table-of-contents).

**Reply by stereobooster** [reply]
> Removing quotes from block quotes makes sense also, although I feel like that's more an upstream issue with Tailwind typography

```js
module.exports = {
  theme: {
    extend: {
      typography: {
        quoteless: {
          css: {
            'blockquote p:first-of-type::before': { content: 'none'},
            'blockquote p:first-of-type::after': { content: 'none' },
          },
        },
      },
    },
  },
  variants: {},
  plugins: [require('@tailwindcss/typography')],
}
```

I can do a PR for that (if you want)
